server/AuthController.js

```diff
@@ -4,18 +4,21 @@ let {method} = require('sails-async')
 let F = require('futil-js')
 
 module.exports = {
-  login: (login, username='email', password='password', id='id') => method(async (req, res) => {
-    let user = await login(req.param(username), req.param(password))
-    if (user.error) {
-      if (user.error) F.throws(user.message || user.error)
-      return 401
+  login: (authenticate) => method(async (req, res) => {
+    let userId = await authenticate(req.allParams(), {req, res})
+    if (!_.isNil(userId)) {
+      let token = JWT.issue({
+        user: userId
+      })
+      // Set here so client auto uses it.
+      res.set(JWT.renewTokenHeader, token)
+      return {token}
+    } else {
+      F.throws({
+        statusCode: 403,
+        message: 'Invalid user id.'
```

Comment:
This should _probably_ be a `400` since it's a bad request and not really a forbidden thing